lib/palindrome_check.rb

```diff
@@ -1,7 +1,25 @@
 # A method to check if the input string is a palindrome.
 # Return true if the string is a palindrome. Return false otherwise.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n), where n is determined by the size of the string "my_phrase"
+# Space complexity: O(1), where 1 is the space used by the variables containing the
+# size of the string and the number representing iterations
 def palindrome_check(my_phrase)
-  raise NotImplementedError
+  if my_phrase.class != String
+    return false
+  end
+  i = 0
+  x = my_phrase.length
+  while i <= x
```

Comment:
[nit] If you make this `i < x` instead, you'll save one unnecessary loop iteration.